pytensor/tensor/rewriting/linalg.py

```diff
@@ -377,3 +379,55 @@ def local_lift_through_linalg(
                 return [block_diag(*inner_matrices)]
             else:
                 raise NotImplementedError  # pragma: no cover
+
+
+# Det Diag Rewrite
+@register_canonicalize
+@register_stabilize
+@node_rewriter([det])
+def det_diag_rewrite(fgraph, node):
```

Comment:
It's helpful to add some representation of the rewrite in the docstrings. Something like `det(eye(5) * vec_x) -> prod(vec_x)` (and also the case for the scalar_x with the power)